blues/simulation.py

```diff
@@ -8,263 +8,647 @@
+        #Construct MD Integrator and Simulation
+        self.integrator = self.generateIntegrator(**self.config)
 
+        #Check for pressure parameter to set simulation to NPT
+        if 'pressure' in self.config.keys():
+            self._system = self.addBarostat(self._system, **self.config)
+            logger.warning('NCMC simulation will NOT have pressure control. NCMC will use pressure from last MD state.')
+        else:
+            logger.info('MD simulation will be {} NVT.'.format(self._config['temperature']))
+        self.md = self.generateSimFromStruct(self._structure, self._system, self.integrator, **self.config)
 
-class Simulation(object):
-    """Simulation class provides the functions that perform the BLUES run.
+        #Alchemical Simulation is used for computing correction term from MD simulation.
+        alch_integrator = self.generateIntegrator(**self.config)
+        self.alch = self.generateSimFromStruct(self._structure, self._system, alch_integrator, **self.config)
+
+        #Construct NCMC Integrator and Simulation
+        self.ncmc_integrator = self.generateNCMCIntegrator(**self.config)
 
-    Ex.
-        import blues.ncmc
-        blues = ncmc.Simulation(sims, move_engine, **opt)
-        blues.run()
+        #Initialize the Move Engine with the Alchemical System and NCMC Integrator
+        for move in self._move_engine.moves:
+            self._alch_system, self.ncmc_integrator = move.initializeSystem(self._alch_system, self.ncmc_integrator)
+        self.ncmc = self.generateSimFromStruct(self._structure, self._alch_system, self.ncmc_integrator, **self.config)
+        SimulationFactory.print_host_info(self.ncmc)
 
+class Simulation(object):
+    """Simulation class provides the functions that perform the BLUES run.
     """
-    def __init__(self, simulations, move_engine, **opt):
+    def __init__(self, simulations):
         """Initialize the BLUES Simulation object.
 
         Parameters
         ----------
-        sims : blues.ncmc.SimulationFactory object
+        simulations : blues.ncmc.SimulationFactory object
             SimulationFactory Object which carries the 3 required
             OpenMM Simulation objects (MD, NCMC, ALCH) required to run BLUES.
-        move_engine : blues.ncmc.MoveEngine object
-            MoveProposal object which contains the dict of moves performed
-            in the NCMC simulation.
-        """
-        if 'Logger' in opt:
-            self.log = opt['Logger']
-        elif simulations.log:
-            self.log = simulations.log
-        else:
-            self.log = logging.getLogger(__name__)
-
-        self.opt = opt
-        self.md_sim = simulations.md
-        self.alch_sim = simulations.alch
-        self.nc_sim = simulations.nc
-        self.move_engine = move_engine
 
+        """
+        self._config = simulations.config
+        self._move_engine = simulations._move_engine
+        self._md_sim = simulations.md
```

Comment:
I don't know why this didn't occur to me way earlier, but it'd be handy to have an explicit method to get the md and ncmc simulations attached to the BLUES `Simulation`. That way if someone wants to do something related to those things, like add another reporter, it'd be straightforward to access and change without knowing the object's attributes.

So something super simple like:
```
def getMDSimulation(self):
    return self._md_sim
def getNCMCSimulation(self):
    return self._ncmc_sim
```